src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AddUsersStep.ts

```diff
@@ -0,0 +1,153 @@
+import pluralize from 'pluralize'
+import { ReactNode, useState } from 'react'
+import { div, h, p } from 'react-hyperscript-helpers'
+import { useUniqueId } from 'src/components/common'
+import { ValidatedInput } from 'src/components/input'
+import { formHint, FormLabel } from 'src/libs/forms'
+import * as Utils from 'src/libs/utils'
+import {
+  columnEntryStyle,
+  columnStyle, hintStyle,
+  rowStyle
+} from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/styles'
+import {
+  LabeledRadioButton, LabeledRadioGroup
+} from 'src/pages/billing/NewBillingProjectWizard/StepWizard/LabeledRadioButton'
+import { Step } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/Step'
+import {
+  legendDetailsStyle,
+  StepFieldLegend,
+  StepFields
+} from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepFields'
+import { StepHeader } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepHeader'
+import { validate } from 'validate.js'
+
+
+interface EmailInputProps {
+  emails: string
+  onSetEmails: (emails: string, hasError: boolean) => void
+  disabled: boolean
+  errors: ReactNode
+  setErrors: (ReactNode) => void
+  onFocus: () => void
+  label: string
+  addSpacing: boolean
+}
+
+const EmailInput = (props: EmailInputProps) => {
+  const inputId = useUniqueId()
+
+  const getIndividualEmails = emails => emails?.split(',').map(email => email.trim()).filter(email => !!email)
+  const validateEmails = individualEmails => {
+    const errors = individualEmails
+      .map(email => !!email && Utils.summarizeErrors(validate({ [email]: email }, { [email]: { email: true } })?.[email]))
+      .filter(error => !!error)
+    return errors
+  }
+
+  return div({ style: columnEntryStyle(props.addSpacing) }, [
+    h(FormLabel, { htmlFor: inputId }, [props.label]),
+    h(ValidatedInput, {
+      inputProps: {
+        id: inputId,
+        placeholder: 'comma-separated email addresses',
+        disabled: props.disabled,
+        onChange: emails => {
+          const individualEmails = getIndividualEmails(emails)
+          const emailErrors = validateEmails(individualEmails)
```

Comment:
@blakery added debouncing, and it feels natural to me.

Reply:
@aherbst-broad you must not have the most recent version of the code-- check that you have the last commit.

Reply:
I don't think we want to wait until focus leaves the field because then you don't get the feedback until you move to work on something else. The delayed validation felt good to me.